main/src/cgeo/geocaching/maps/mapsforge/v6/layers/ThunderforestMap.java

```diff
@@ -0,0 +1,39 @@
+package cgeo.geocaching.maps.mapsforge.v6.layers;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import org.mapsforge.core.model.Tile;
+import org.mapsforge.map.layer.download.tilesource.AbstractTileSource;
+
+/**
+ * Like {@link org.mapsforge.map.layer.download.tilesource.OpenCycleMap} but loading tiles from thunderforest servers with apikey.
+ */
+public class ThunderforestMap extends AbstractTileSource {
+
+    public static final ThunderforestMap INSTANCE = new ThunderforestMap(new String[]{"a.tile.thunderforest.com", "b.tile.thunderforest.com", "c.tile.thunderforest.com"}, 443);
+
+    public ThunderforestMap(final String[] hostNames, final int port) {
+        super(hostNames, port);
+    }
+
+    public int getParallelRequestsLimit() {
+        return 8;
+    }
+
+    public URL getTileUrl(final Tile tile) throws MalformedURLException {
+        return new URL("https", this.getHostName(), this.port, "/cycle/" + tile.zoomLevel + '/' + tile.tileX + '/' + tile.tileY + ".png?apikey=d6866308a6e24783822338c9a49ceb6c");
```

Comment:
I suggest to remove the key similarly to all other keys we use:
* create a new resource string in keys.xml (and in its template version), including a comment where to register for a new key
* use the resource string in the code
* have @mucek4 put the actual value into the file on the CI server